web/pdf_document_properties.js

```diff
@@ -29,32 +31,69 @@ class PDFDocumentProperties {
   /**
    * @param {PDFDocumentPropertiesOptions} options
    */
-  constructor(options) {
-    this.overlayName = options.overlayName;
-    this.fields = options.fields;
-    this.container = options.container;
+  constructor({ overlayName, fields, container, closeButton, }) {
+    this.overlayName = overlayName;
+    this.fields = fields;
+    this.container = container;
 
-    this.rawFileSize = 0;
-    this.url = null;
-    this.pdfDocument = null;
+    this._reset();
 
-    // Bind the event listener for the Close button.
-    if (options.closeButton) {
-      options.closeButton.addEventListener('click', this.close.bind(this));
+    if (closeButton) {
+      // Bind the event listener for the Close button.
+      closeButton.addEventListener('click', this.close.bind(this));
     }
-    this._dataAvailableCapability = createPromiseCapability();
-
     OverlayManager.register(this.overlayName, this.container,
                             this.close.bind(this));
   }
 
   /**
    * Open the document properties overlay.
+   *
+   * Note that since the document properties are fetched asynchronously,
+   * the dialog will *only* be updated if it's still open when the data arrives.
```

Comment:
> If the dialog is closed (not visible), then it will still be updated in the background for the next time you open it.

That should be the opposite of what the code does, both currently and with this patch.
Since we always re-fetch the data, and update the dialog every time it's opened, there's no point to updating anything if the dialog has been closed by the time e.g. `this.pdfDocument.getMetadata()` is resolved. (Keep in mind that all the operations involved here are asynchronous.)

So basically, I was trying to provide a general comment for these checks:
```javascript
if (OverlayManager.active !== this.overlayName) {
  return;
}
```

> Open https://mozilla.github.io/pdf.js/web/viewer.html, wait two seconds and open the document properties and you'll immediately see the data.

Noting gets loaded until the dialog is successfully opened, so what you describe as "immediately" just *seem* instantaneous since two seconds after loading the `tracemonkey.pdf` file, there's probably not many heavy operations still occurring. So the asynchronous fetching of the various document properties thus becomes very fast.

> Or is that not what you meant here?

No, that's not really what I meant :-)
So any suggestions on making this easier to understand would be helpful!
(I could also move, or even just remove, the comment if doesn't help make sense of the code.)